helix-lock/src/main/java/org/apache/helix/lock/DistributedLock.java

```diff
@@ -49,4 +49,9 @@
    * false if the user is not the lock owner or the lock doesn't have a owner
    */
   boolean isCurrentOwner();
+
+  /**
+   * Close the lock and release resources
+   */
+  void close();
```

Comment:
So we would expect lock users to explicitly call this one? This is a bit unintuitive, when put together with the other APIs that are intended for lock purpose only.

Reply:
Discussed offline, for now we will let user to explicitly call the close method, since we cannot determine all the use cases where we should close the lock right now. After we have a clearer picture of all the features, we can try to do this for our users.